src/Nimiq/Address.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright © 2017-2019 Trust.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Address.h"
+
+#include "../Hash.h"
+#include "../HexCoding.h"
+
+#include <TrezorCrypto/base32.h>
+#include <TrezorCrypto/blake2b.h>
+
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+
+using namespace TW::Nimiq;
+
+static const char *BASE32_ALPHABET_NIMIQ = "0123456789ABCDEFGHJKLMNPQRSTUVXY";
+
+static void check_append(std::string &, char);
+
+bool Address::isValid(const std::string& stringPadded) {
+    // Magic check
+    if (stringPadded.substr(0, 2) != "NQ")
+        return false;
+
+    std::string string = stringPadded;
+
+    // Remove spaces
+    string.erase(
+        std::remove(string.begin(), string.end(), ' '),
+        string.end());
+
+    if (string.length() != 36)
+        return false;
+
+    // Check if valid Base32
+    auto hash = std::array<uint8_t, Address::size>();
+    if (base32_decode(string.data() + 4, 32, hash.data(), hash.size(), BASE32_ALPHABET_NIMIQ) == NULL)
+        return false;
+
+    // Build checksum state
+    std::string check = "";
+    for (int i = 4; i < 36; i++)
+        check_append(check, string[i]);
+    check.append("232600");
+
+    // Compress checksum state
+    std::string tmp;
+    for (int i = 0; i < (int)(std::ceil(check.length() / 6.0)); i++)
+        tmp = std::to_string(std::stoi(tmp + check.substr(i * 6, 6)) % 97);
+    int checksum_should = 98 - std::stoi(tmp);
```

Comment:
Why does `tmp` need to be a string? Looks like it should be an `int`. Also let's come up with a better name, `tmp` doesn't tell me what it's for.

Reply:
Removed `tmp`.